plugins/module_utils/foreman_helper.py

```diff
@@ -444,28 +447,32 @@ def _delete_entity(self, resource, current_entity, params):
         payload = {'id': current_entity['id']}
         if params:
             payload.update(params)
-        self.resource_action(resource, 'destroy', payload)
+        self.resource_action(resource, 'destroy', payload, synchronous=synchronous)
         return True, None
 
-    def resource_action(self, resource, action, params, options=None, data=None, files=None):
+    def resource_action(self, resource, action, params, options=None, data=None, files=None, synchronous=True):
         resource_payload = self.foremanapi.resource(resource).action(action).prepare_params(params)
         if options is None:
             options = {}
         try:
             result = None
             if not self.check_mode:
                 result = self.foremanapi.resource(resource).call(action, resource_payload, options=options, data=data, files=files)
+                if synchronous and isinstance(result, dict) and 'action' in result and 'state' in result and 'pending' in result:
+                    # looks like a task
```

Comment:
Nitpick: if you have to put a comment in a complicated condition like this, why not make a boolean variable that explains such?

```rb
is_foreman_task = synchronous and isinstance(result, dict) and 'action' in result and 'state' in result and 'pending' in result
if is_foreman_task:
  ...
```

Reply:
I am hesitant to define a variable to use it exactly once, when a little comment in the right place can provide the same information.

Reply:
Never fear, your variable used only once will be optimized away, I'm sure. DRY is not the only reason to use a variable IMO.

Reply:
changing it. ;)